apps/arweave/src/ar_nonce_limiter.erl

```diff
@@ -1232,6 +1289,34 @@ debug_double_check(Label, Result, Func, Args) ->
 			end
 	end.
 
+filter_step_triplets([], _LowerBounds) ->
+	[];
+filter_step_triplets([{O, _, _} = Triplet | Triplets], LowerBounds) ->
+	case lists:member(O, LowerBounds) of
+		true ->
+			[];
+		false ->
+			[Triplet | filter_step_triplets(Triplets, LowerBounds)]
+	end.
+
+get_triplets(_StepNumber, _Steps, _ResetPoint, _UpperBound, _NextUpperBound, 0) ->
+	[];
+get_triplets(_StepNumber, [], _ResetPoint, _UpperBound, _NextUpperBound, _N) ->
+	[];
+
+get_triplets(StepNumber, [Step | Steps], ResetPoint, UpperBound, NextUpperBound, N) ->
```

Comment:
Is there any chance, if N is big enough, that we might through 2 reset points?

Reply:
We only consider one VDF session here and every session has one reset point (the "first" one).